op-program/client/program.go

```diff
@@ -46,23 +46,12 @@ func RunProgram(logger log.Logger, preimageOracle io.ReadWriter, preimageHinter
 	l1PreimageOracle := l1.NewCachingOracle(l1.NewPreimageOracle(pClient, hClient))
 	l2PreimageOracle := l2.NewCachingOracle(l2.NewPreimageOracle(pClient, hClient))
 
-	bootInfo := NewBootstrapClient(pClient).BootInfo()
-	logger.Info("Program Bootstrapped", "bootInfo", bootInfo)
-	safeHead, outputRoot, err := tasks.RunDerivation(
-		logger,
-		bootInfo.RollupConfig,
-		bootInfo.L2ChainConfig,
-		bootInfo.L1Head,
-		bootInfo.L2OutputRoot,
-		bootInfo.L2ClaimBlockNumber,
-		l1PreimageOracle,
-		l2PreimageOracle,
-	)
-	if err != nil {
-		return err
+	bootInfo := boot.NewBootstrapClient(pClient).BootInfo()
+	if os.Getenv("USE_INTEROP") == "true" {
+		return interop.RunInteropProgram(logger, bootInfo, l1PreimageOracle, l2PreimageOracle)
 	}
 	if flags == RunProgramFlagValidate {
-		return claim.ValidateClaim(logger, safeHead, eth.Bytes32(bootInfo.L2Claim), outputRoot)
+		return RunPreInteropProgram(logger, bootInfo, l1PreimageOracle, l2PreimageOracle)
```

Comment:
We need to split up derivation and claim validation into two separate tasks.

The usecase for `RunProgramFlagValidate` and `RunProgramFlagSkipValidation` is to allow the prefetcher to run the client without validating the prestate. The flag is pretty much useless here in the latter case as the op-program does nothing other than boot up.

Reply:
Yeah I think a merge went weird here. I fixed it in the next PR and was confused why you put the flag there but I must have merged it wrong at some point.  I'll pull the fix back into this PR so we keep things sane all the way along.

Reply:
Have only implemented skipping validation when interop is enabled since non-interop shouldn't need it and I'm keen to keep things as feature toggled as possible for safety.  Will see if that breaks tests...